faux-mgs/src/main.rs

```diff
@@ -1818,6 +1829,40 @@ async fn run_command(
             let result = sp.read_host_flash(slot, addr).await?;
             Ok(Output::Lines(vec![format!("{result:x?}")]))
         }
+        Command::DumpHostFlash { slot, size, output } => {
+            let mut fh = OpenOptions::new()
+                .write(true)
+                .create_new(true)
+                .open(&output)
+                .with_context(|| format!("failed to create `{output}`"))?;
+
+            let log_every = Duration::from_secs(5);
+            let mut last_log = Instant::now();
+            info!(log, "starting to dump host flash");
+
+            let mut addr = 0;
+            while addr < size {
+                if last_log.elapsed() > log_every {
+                    info!(
+                        log, "continuing to dump host flash";
+                        "offset" => addr,
+                    );
+                    last_log = Instant::now();
+                }
+                let data = sp.read_host_flash(slot, addr).await.with_context(
+                    || format!("failed to read data at offset {addr}"),
+                )?;
+                let to_write = usize::min(data.len(), (size - addr) as usize);
+                fh.write_all(&data[..to_write])
+                    .with_context(|| format!("failed writing to `{output}`"))?;
+                addr += to_write as u32;
+            }
```

Comment:
consider https://docs.rs/indicatif/latest/indicatif/ for progress? We use this in humility for progress bars